### PR TITLE
Update Python and Django versions

### PR DESCRIPTION
**Pull request by adamchainz**
This PR does a bunch of updates:

1. Drop old Python and Django versions.
2. Add Python 3.11 and 3.12.
3. Add Django 4.2 and 5.0.
4. Upgrade the actions on GitHub Actions.
5. Move flake8 and isort to run with the latest Python.
6. Upgrade isort.

It also standardizes the testing with other projects a little bit, see comments.

All tests passed locally for me with `tox -p 5`. The change to use `-Werror` means there are no warnings even.

### CODE REVIEW

**Comment by charettes** [comment]
I unfortunately don't have administrative rights on the repository @adamchainz so I can't grant access for worklows to always run. I'll keep monitoring your pushes until they pass.

File: tox.ini [hunk 1]
```diff
@@ -2,50 +2,46 @@
...
+    3.11: py311
+    3.12: py312
 
 [testenv]
-basepython =
-    py36: python3.6
-    py37: python3.7
-    py38: python3.8
-    py39: python3.9
-    py310: python3.10
```
[the remark this answers is not in the document]
Comment (charettes) [reply tox.ini:27]: I think this is artifact from of old TravisCI or Tox versions.

You've confirmed that `py3X` automatically maps to `python3.X` locally?

**Comment by charettes** [comment]
@adamchainz thinks a looking good, I'll keep the PR unmerged for now as a reminder to merge + publish a new version by EOD.

**Comment by charettes** [comment]
[I just published 3.2.0](https://pypi.org/project/django-picklefield/3.2/). You should be all set.

**Comment by adamchainz** [comment]
Thank you! I realized I missed `python_requires`, so I made #66 as a follow-up.

**Pull request merged**